CircleCI Configuration

TaysirTayyab opened the pull request:
This PR is pretty straightforward. It adds the CircleCI configuration file so that we can test this project on CircleCI.

The `README` has also been updated to also describe setup for people running Docker for Mac (where as before it only applied to people  running Docker Toolbox).

The repository will need to be followed in CircleCI and the checks will need to be setup to make sure sure nothing can be merged unless the tests pass.

TaysirTayyab commented:
!reassign testing @Chekote

medology-bugman commented:
Paging @Chekote to review the testing work in this pull request!

Chekote commented on bin/php:
```diff
@@ -1,8 +1,8 @@
 done
 
-docker run -it -v $(pwd):/app -w /app php:cli php $ARGS
+docker run --rm -it -v $(pwd):/app --link web:dockermachine.local --net flexiblemink_default -w /app php:cli php $ARGS
```
FYI : The php:cli tag is frequently updated, so you will be subject to external changes that could make your tests unstable.

TaysirTayyab replied:
I added a version to it and did the same for Apache.

Chekote commented on circle.yml:
```diff
@@ -0,0 +1,24 @@
+
+  post:
+    - mv artifacts/default.xml ${CIRCLE_TEST_REPORTS}/behat.junit.xml
+    - mv artifacts/phpunit.junit.xml ${CIRCLE_TEST_REPORTS}/phpunit.junit.xml
```
Please add a line break at the end of this file.

Chekote commented on circle.yml:
```diff
@@ -0,0 +1,24 @@
+  override:
+    - sudo pip install docker-compose
+    - docker-compose -p flexiblemink -f docker/docker-compose.yml up -d
+    - bin/init_project
```
Since the docker-compose command has the -d argument, you may want to put a small delay after it. There's a chance that the init_project script will try to access one of the containers before it's ready.

Chekote commented:
Testing review for 913eca0 is complete.

TaysirTayyab commented:
@Chekote this is ready for review again.

- cleaned up the `$ARGS` section of scripts for now. When we need them, they can be added back in, but for now no point in bloat
- added version to `httpd` and `php-cli` images to keep from inadvertent breakage from the `latest` tag
- took care of that line break in the `circle.yml`

Chekote commented:
795df83 is good to merge 👍
